source/acquaman/AMScanActionControllerBasicFileWriter.cpp

```diff
@@ -62,6 +65,12 @@ AMScanActionControllerBasicFileWriter::AMScanActionControllerBasicFileWriter(con
 		if(!spectraFile_->open(QIODevice::WriteOnly | QIODevice::Text))
 			emitError(AMScanActionControllerBasicFileWriter::CouldNotOpenError);
 	}
+
+	// start the timer (2mins) to check whether we are writing to the files consistently
+	QTimer *dataFileSizeCheckTimer = new QTimer(this);
```

Comment:
@davidChevrier @dretrex Yes, we need to call deleteLater() for the fileWriter since it is NOT a child of the parent class. Therefore, it has to be destroyed explicitly.

Maybe I'm wrong. But QT documented that QObject can destroy its child automatically. That's the reason I use `new QTimer(this)` instead of `new QTimer()` here. I'm expecting the super class can destroy the timer for me.